Fuse color sections for named-color

teoli2003 opened the pull request:
This fuses the different sections for `named-color`:
- CSS 1 / CSS 2 / CSS 3 color lists predate Firefox 1 and Chrome 1. No developer has to deal with compatibility problem here.
- It is easier to browse a complete list, then to have to look in 4 different sections.

teoli2003 committed:
Fuse color sections for named-color

teoli2003 committed:
Fix Markdownlint

wbamberg commented on files/en-us/web/css/named-color/index.md:
```diff
@@ -27,126 +27,11 @@ color: transparent;
-### CSS Level 3 values
-
-Although various colors not in the specification (mostly adapted from the X11 colors list) were supported by early browsers, it wasn't until SVG 1.0 and [CSS Colors Level 3](https://drafts.csswg.org/css-color-3/) that they were formally defined. They are called the _extended color keywords_, the _X11 colors_, or the _SVG colors_.
+The following colors have a keyword associated to them:
```
```suggestion
The following colors have a keyword associated with them:
```

wbamberg reviewed (changes requested):
This makes sense to me.

Maybe just me, but I think centre-aligning makes this harder to read as well as making the table markup more complicated. cf:

<img width="758" alt="Screen Shot 2023-02-15 at 10 59 55 AM" src="https://user-images.githubusercontent.com/432915/219127031-b3fca6e9-a883-4351-910d-3324d4c89aa8.png">

Josh-Cena commented:
FWIW, I slightly prefer separate lists. CSS 1 colors include all the "modal colors" that I can quickly grab for prototyping to ensure sufficient contrast. CSS 3 colors are all the more obscure names that I don't usually need. The "SVG colors" name is also occasionally seen. CSS 4 also includes the interesting history which now seems lost.

wbamberg commented:
> FWIW, I slightly prefer separate lists. CSS 1 colors include all the "modal colors" that I can quickly grab for prototyping to ensure sufficient contrast. CSS 3 colors are all the more obscure names that I don't usually need. The "SVG colors" name is also occasionally seen and we can teach people what it means. CSS 4 also includes the interesting history which now seems lost.

Yeah, I think there's merit in that. @teoli2003 , what do you think?

(I still don't like the centering)

teoli2003 commented:
I'm still pondering here. I agree Josh'es point is valuable, and my current thought is:

- Just before the list, add a link, or some text, to a tool (or a something else) to select _accessible_ colors (that are high contrast), with an additional comment remembering that no information should be conveyed only by color. 
- The merged list (without centering)
- A paragraph/section about the history of color names, from X11 to CSS Colors L4. [Interesting for the spec aficionados, but probably not for the web devs]

I'm still thinking about how to do the first part quickly and efficiently. Also, I don't know if using the CSS1 colors would solve the accessibility problem of choosing colors; likely not. (That's why, I'm not yet convinced)

As I said, I'm still pondering here.

Josh-Cena commented:
It's mostly not about accessibility, just that I don't have to choose over the entire $256^3$ RGB space and quickly get evenly spread out colors for prototyping. Fewer choices mean faster decision.

teoli2003 committed:
Don't center text in table

teoli2003 committed:
Single out 16 colours

teoli2003 committed:
Write a new history section

teoli2003 commented:
So I did the following:
- I singled out the 16 colors (I also kept them in the extensive list). That should solve the use case of looking for colors with high contrast.
- I added 2 paragraphs about the history after the tables, I think this is more coherent. You read the whole history in one place.
- I removed the "centering" of the text. It is more coherent with the Markdown tables.

estelle commented on files/en-us/web/css/named-color/index.md:
```diff
@@ -158,701 +137,774 @@ Although various colors not in the specification (mostly adapted from the X11 co
+    </tr>
+    <tr>
+      <td>
+        <a href="https://codepen.io/trezy/post/honoring-a-great-man"
```
```suggestion
        <a href="https://meyerweb.com/eric/thoughts/2014/06/19/rebeccapurple/"
```

So we don't have a duplicate link.

estelle replied:
let's switch the other one, since the other one mentions him by name so we should link to his blog.

teoli2003 replied:
So I'm keeping this post here, and change the link to the blog at the bottom

estelle commented on files/en-us/web/css/named-color/index.md:
```diff
@@ -158,701 +137,774 @@ Although various colors not in the specification (mostly adapted from the X11 co
-      <td style="text-align: center"><code>grey</code></td>
-      <td><code>#808080</code></td>
+      <td><code>grey</code></td>
+      <td><code>#808080</code></td> (synonym of <code>gray</code>)
```
```suggestion
      <td><code>#808080</code> (synonym of <code>gray</code>)</td>
```

estelle commented on files/en-us/web/css/named-color/index.md:
```diff
@@ -158,701 +137,774 @@ Although various colors not in the specification (mostly adapted from the X11 co
 </table>
 
-#### transparent
+Initially, in [CSS Level 1](https://www.w3.org/TR/REC-CSS1/), only 16 basic colors were defined, with `orange` added in [CSS Level 2](https://www.w3.org/TR/CSS2/). Web designers found this list too short, and browser vendors added numerous names for colors based on the X11 color names. In [SVG 1.0](https://www.w3.org/TR/2001/REC-SVG-20010904/), then in [CSS Colors Level 3](https://www.w3.org/TR/c ...
```
```suggestion
Initially, in [CSS Level 1](https://www.w3.org/TR/REC-CSS1/), only 16 basic colors were defined, with `orange` added in [CSS Level 2](https://www.w3.org/TR/CSS2/). Web designers found this list too short, and browser vendors added numerous names for colors based on the X11 color names. In [SVG 1.0](https://www.w3.org/TR/2001/REC-SVG-20010904/), then in [CSS Colors Level 3](https://www.w3.org/TR/css-color-3/), these names got standardized, formally defined, and made uniform (some had different spellings that are now aliases). They are called _extended color keywords_, _X11 colors_, or _SVG colors_.
```

estelle commented on files/en-us/web/css/named-color/index.md:
```diff
@@ -158,701 +137,774 @@ Although various colors not in the specification (mostly adapted from the X11 co
-#### transparent
+Initially, in [CSS Level 1](https://www.w3.org/TR/REC-CSS1/), only 16 basic colors were defined, with `orange` added in [CSS Level 2](https://www.w3.org/TR/CSS2/). Web designers found this list too short, and browser vendors added numerous names for colors based on the X11 color names. In [SVG 1.0](https://www.w3.org/TR/2001/REC-SVG-20010904/), then in [CSS Colors Level 3](https://www.w3.org/TR/c ...
+
+In [CSS Colors Level 4](https://www.w3.org/TR/css-color-4/), an additional color, `rebeccapurple` was added [to honor web pioneer Eric Meyer](https://codepen.io/trezy/post/honoring-a-great-man), who suffered a terrible personal tragedy.
```
```suggestion
In [CSS Colors Level 4](https://www.w3.org/TR/css-color-4/), an additional color, `rebeccapurple` was added to honor [web pioneer Eric Meyer](https://meyerweb.com/eric/thoughts/2014/06/19/rebeccapurple/), who suffered a terrible personal tragedy.
```

estelle commented:
The spec reads "(Two special color values, [transparent](https://w3c.github.io/csswg-drafts/css-color/#valdef-color-transparent) and [currentcolor](https://w3c.github.io/csswg-drafts/css-color/#valdef-color-currentcolor), are specially defined in their own sections.)" We mention transparent here, but not currentColor.

teoli2003 committed:
Update files/en-us/web/css/named-color/index.md

Co-authored-by: Estelle Weyl <estelle@weyl.org>

teoli2003 committed:
Update files/en-us/web/css/named-color/index.md

Co-authored-by: Estelle Weyl <estelle@weyl.org>

teoli2003 committed:
Update files/en-us/web/css/named-color/index.md

Co-authored-by: Estelle Weyl <estelle@weyl.org>

teoli2003 commented on files/en-us/web/css/named-color/index.md:
```diff
@@ -27,9 +27,11 @@ color: transparent;
 ### Value
 
-#### CSS Level 1 values
+Named colors consists of standard colors, and of the `transparent` keyword.
```
```suggestion
Named colors consists of standard colors, the `transparent` and [`currentcolor`](/en-US/docs/Web/CSS/color_value#currentcolor_keyword) keywords.
```

teoli2003 committed:
Update files/en-us/web/css/named-color/index.md

teoli2003 commented:
> The spec reads "(Two special color values, [transparent](https://w3c.github.io/csswg-drafts/css-color/#valdef-color-transparent) and [currentcolor](https://w3c.github.io/csswg-drafts/css-color/#valdef-color-currentcolor), are specially defined in their own sections.)" We mention transparent here, but not currentColor.

Added. Note that the prose and the syntax in the spec technically diverge, but it is worth mentioning.
![Capture d’écran 2023-03-31 à 10 49 34](https://user-images.githubusercontent.com/1466293/229074541-8cefd765-bcee-477b-b8b9-048081046c31.png)

teoli2003 commented:
@wbamberg , @estelle. This is ready for a new pass.

teoli2003 committed:
Uniformize synonyms

teoli2003 committed:
Link to transparent info

estelle commented on files/en-us/web/css/named-color/index.md:
```diff
@@ -158,701 +137,774 @@ Although various colors not in the specification (mostly adapted from the X11 co
+    </tr>
+    <tr>
+      <td>
+        <a href="https://codepen.io/trezy/post/honoring-a-great-man"
+          ><code>rebeccapurple</code></a
+        >
+      </td>
```
```suggestion
        <code>rebeccapurple</code>
      </td>
```

estelle replied:
Let's just get rid of this here as it's confusing when only one color has a link.

estelle commented on files/en-us/web/css/named-color/index.md:
```diff
@@ -158,701 +137,774 @@ Although various colors not in the specification (mostly adapted from the X11 co
-#### transparent
+Initially, in [CSS Level 1](https://www.w3.org/TR/REC-CSS1/), only 16 basic colors were defined, with `orange` added in [CSS Level 2](https://www.w3.org/TR/CSS2/). Web designers found this list too short, and browser vendors added numerous names for colors based on the X11 color names. In [SVG 1.0](https://www.w3.org/TR/2001/REC-SVG-20010904/), then in [CSS Colors Level 3](https://www.w3.org/TR/c ...
+
+In [CSS Colors Level 4](https://www.w3.org/TR/css-color-4/), an additional color, `rebeccapurple` was added to honor [web pioneer Eric Meyer](https://meyerweb.com/eric/thoughts/2014/06/19/rebeccapurple/), who suffered a terrible personal tragedy.
```
```suggestion
In [CSS Colors Level 4](https://www.w3.org/TR/css-color-4/#named-colors), an additional color, `rebeccapurple` was added to honor [web pioneer Eric Meyer](https://meyerweb.com/eric/thoughts/2014/06/19/rebeccapurple/), who suffered a terrible personal tragedy.
```

estelle commented on files/en-us/web/css/named-color/index.md:
```diff
@@ -158,701 +137,774 @@ Although various colors not in the specification (mostly adapted from the X11 co
+
+In [CSS Colors Level 4](https://www.w3.org/TR/css-color-4/), an additional color, `rebeccapurple` was added to honor [web pioneer Eric Meyer](https://meyerweb.com/eric/thoughts/2014/06/19/rebeccapurple/), who suffered a terrible personal tragedy.
+
+### transparent
```
should this be a capital letter?

teoli2003 replied:
No this is the keyword

teoli2003 committed:
Update files/en-us/web/css/named-color/index.md

Co-authored-by: Estelle Weyl <estelle@weyl.org>

teoli2003 committed:
Update files/en-us/web/css/named-color/index.md

Co-authored-by: Estelle Weyl <estelle@weyl.org>

teoli2003 committed:
Update files/en-us/web/css/named-color/index.md

Co-authored-by: Estelle Weyl <estelle@weyl.org>

teoli2003 committed:
Update index.md

wbamberg reviewed (approved):
👍 thank you @teoli2003 !
